pallets/confidential-asset/src/lib.rs

```diff
@@ -370,20 +404,247 @@ decl_module! {
 
             Ok(())
         }
+
+        /// Resets the `FailedOutgoingBalance` and `IncomingBalance` accumulators for the caller's account.
+        /// If successful, the account owner must use their current balance minus the sum of all unsettled outgoing
+        /// balances as their pending balance.
+        ///
+        /// # Arguments
+        /// * `origin` - contains the secondary key of the caller (i.e who signed the transaction to execute this function).
+        /// * `account_id` - the mercat account ID of the `origin`.
+        ///
+        /// # Errors
+        /// - `BadOrigin` if not signed.
+        ///
+        /// # Weight
+        /// `3_000_000_000`
+        #[weight = 3_000_000_000]
+        pub fn update_account_balance_checkpoint(
+            origin,
+            account_id: MercatAccountId,
+        ) -> DispatchResult {
+            let owner = ensure_signed(origin)?;
+            let owner_did = Context::current_identity_or::<Identity<T>>(&owner)?;
+
+            Self::reset_mercat_pending_state(&owner_did, &account_id);
+
+            // Emit an event and include account's current balance for account owner's information.
+            let current_balance =
+                Self::mercat_account_balance(owner_did, account_id.clone());
+            Self::deposit_event(RawEvent::ConfidentialAccountNewCheckpoint(
+                owner_did,
+                account_id,
+                current_balance
+            ));
+
+            Ok(())
```

Comment:
```suggestion
```

Reply:
Substrate has this special thing where you can omit `Ok(())` for extrinsics if you also remove `-> DispatchResult`; it will still return `DispatchResult` but you don't have to write it out.